src/proto/streams/recv.rs

```diff
@@ -531,6 +570,34 @@ impl Recv {
         Ok(())
     }
 
+    /// Add a locally reset stream to queue to be eventually reaped.
+    pub fn enqueue_reset_expiration(
+        &mut self,
+        stream: &mut store::Ptr,
+        counts: &mut Counts,
+    ) {
+        assert!(stream.state.is_local_reset());
+
+        if stream.is_pending_reset_expiration() {
+            return;
+        }
+
+        if !counts.can_inc_num_reset_streams() {
+            // try to evict 1 stream if possible
+            // if max allow is 0, this won't be able to evict,
+            // and then we'll just bail after
+            if let Some(evicted) = self.pending_reset_expired.pop(stream.store_mut()) {
+                let is_counted = evicted.is_counted();
```

Comment:
This line was confusing at first. My initial thought was that, at this point, the stream could never be "counted" as it has already been reset. However, a stream transitions to `pending_reset_expiration` immediately when the client resets the stream, but the stream transitions to "not counted" once the RST_STREAM frame is sent. So, there is a period in which the stream is in the "pending reset expired" queue while it is still counted. Under a specific workload, this line could be hit while the stream is still counted.

Could you add some comments describing this?